frontend/src/core/user/components/UserAutoUpdater.js

```diff
@@ -0,0 +1,60 @@
+/* @flow */
+/* globals IntervalID */
+
+import * as React from 'react';
+import { connect } from 'react-redux';
+
+import { actions } from '..';
+
+
+type Props = {
+};
+
+type InternalProps = {
+    ...Props,
+    dispatch: Function,
+};
+
+type State = {|
+    timer: ?IntervalID,
+|};
+
+
+/**
+ * Regularly fetch user data to keep it up-to-date with the server.
+ */
+export class UserAutoUpdaterBase extends React.Component<InternalProps, State> {
+    constructor() {
+        super();
+        this.state = {
+            timer: null,
+        };
+    }
+
+    fetchUserData = () => {
+        this.props.dispatch(actions.get());
+    }
+
+    componentDidMount() {
+        this.fetchUserData();
+        const timer = setInterval(this.fetchUserData, 2 * 60 * 1000);
+        this.setState({ timer });
```

Comment:
I might be missing something, but why is the timer stored as part of state? Setting it here will force a re-render of the component, and even though it renders `null`, IMHO it seems more natural to store the timer reference as a property of the component instance, because state provides no additional value needed to implement this logic.